src/utils/fetch.js

```diff
@@ -134,11 +134,13 @@ export function convertFetchSuccess(url, body, timerId, result) {
 
     if ((result.status < 200 || result.status > 299) && result.status !== 304) {
         convertedPromise = convertedPromise.then((newResult) => {
+            const requestId = parseInt(result.headers.get('x-request-id'), 10);
             log.error(LOG_AREA, 'rejected server response', {
                 url,
                 body,
                 status: newResult.status,
                 response: newResult.response,
+                requestId: isNaN(requestId) ? null : requestId,

```

Comment:
Nitpick. 
What do you think about not including **requestId** at all if it's not part of a header or is nan? 
Will make a error logging payload a little bit smaller.

Reply:
Does the existence of the field (even if empty) make a deep dive through logs easier? I haven't had the need yet, so I don't know the possible shortcomings of our tools :)

My thought process was to aid in inspection capability (because that's what logs are for), but if theres no difference, then performance obviously wins.

Reply:
Yeah, null might be beneficial to detect that we indeed report it but it was empty. Otherwise there will be distinction between old code version which didn't include it at all. 

Let's leave it.